### PR TITLE
Updated scheme of version numbering

### LINKED ISSUE #3037: Scripting URLs in User Guild v1.2 need updating.
### Expected Behaviour
In Chapter "17. Scripting", reference URLs should match the versions of software being used. 

### Actual Behaviour
Footnote 1 is for the Qt5 Scripting Engine, which is fine since it's mentioned in the text.  But there should be an additional footnote for the Qt6 QJSEngine, "https://doc.qt.io/qt-6/qtjavascript.html", which is mentioned in the last paragraph of 17.1, and is the version of the JavaScript engine used by the version of Stellarium that this document is for.

Footnote 3 (https://www.stellarium.org/doc/22.x/scripting.html) in 17.1 leads to a "404 Page Not Found" error.  The URL should be "https://stellarium.org/doc/1.x/scripting.htmlhttps://stellarium.org/doc/1.x/scripting.html"

### Steps to reproduce
Click on the footnote URLs in chapter 17.

### System
* Stellarium version: 1.2
* Operating system: Windows 11, Version 10.0.22621 Build 22621
* Graphics Card: NVIDIA GeForce RTX 2080 Super, Driver Version	31.0.15.2824
* Screen type (if applicable): Resolution, HighDPI, scaling: 1920x1080, 100% scaling.

### Logfile 
N/A

### PR DESCRIPTION
**Pull request by alex-w**
### Description
To reduce the misunderstanding for versioning of Stellarium I've updated scheme of version numbering - it's almost complete automatized now.

Fixes #3037, #3018, #2981 (issue)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,6 +20,10 @@ SET_PROPERTY(GLOBAL PROPERTY USE_FOLDERS ON)
 
 ########### Project name ###########
 PROJECT(Stellarium C CXX)
+
+SET(STELLARIUM_RELEASE_BUILD 0 CACHE BOOL "Set 1 to build as an official release (0 for development snapshots).")
+SET(STELLARIUM_RELEASE_NUMBER 0)
+
 ########### Detect Qt version
 SET(ENABLE_QT6 1 CACHE BOOL "Whether to try building with Qt6. If Qt6 is not found, Qt5 will be used.")
```
Comment (gzotti) [inline CMakeLists.txt:28]: In QtCreator, is there a way to detect/analyze the selected "Kit"? When I select Qt5.12 and run cmake on a fresh clone, it happily finds the Qt6.4 installed in parallel until I set this variable manually.

File: CMakeLists.txt [hunk 2]
```diff
@@ -32,28 +36,35 @@ ENDIF()
...
-# If Qt6, set series number to 1.
-IF(Qt6_FOUND)
-     SET(STELLARIUM_MAJOR 1)
+# two last digits of the current year
+STRING(TIMESTAMP SHORTYEAR "%y")
+MATH(EXPR STELLARIUM_MAJOR_DIGIT "${SHORTYEAR} + 0" OUTPUT_FORMAT DECIMAL)
+# Version of Stellarium has format YY.V.YD, where:
+# -- YY (MAJOR) - two last digits of the year of release
+# -- V  (MINOR) - version of the release (0 is used before first release)
+# -- YD (PATCH) - 0 for releases and the day of the current year for snapshots
```
Comment (gzotti) [inline CMakeLists.txt:45]: What do you call a patch release of 23.4 released on December 30, 2023, and the next on January 15, 2024? 
This looks like 

23.4.364
23.4.015

Comment (alex-w) [reply CMakeLists.txt:45]: No, 23.4.364 & 24.0.15

File: CMakeLists.txt [hunk 3]
```diff
@@ -32,28 +36,35 @@ ENDIF()
...
      -DSTELLARIUM_MAJOR=${STELLARIUM_MAJOR}
      -DSTELLARIUM_MINOR=${STELLARIUM_MINOR}
      -DSTELLARIUM_PATCH=${STELLARIUM_PATCH}
 )
 SET(VERSION "${STELLARIUM_MAJOR}.${STELLARIUM_MINOR}.${STELLARIUM_PATCH}")
-# Public version has format YY.V, where:
-# -- YY - two last digits of the year of release
-# -- V  - version of the release
-# Example: first release in year 2023 has version 1.23.0 and public version 23.1
-MATH(EXPR STELLARIUM_V "${STELLARIUM_PATCH} + 1" OUTPUT_FORMAT DECIMAL)
```
Comment (gzotti) [inline CMakeLists.txt:54]: Yes, this was truly confusing. {0|1}.23.0 or {0|1}.23.1 would have been consistent "internal" numbers for 23.0 or 23.1, respectively.

File: CMakeLists.txt (hunk 1 above)
Comment (alex-w) [reply CMakeLists.txt:28]: Hmm... I don't have described issue in my QtCreator

File: CMakeLists.txt (hunk 3 above)
Comment (alex-w) [reply CMakeLists.txt:54]: I've omitted first {0|1}, because it's not critical.

File: CMakeLists.txt (hunk 2 above)
Comment (alex-w) [reply CMakeLists.txt:45]: The first idea for third number was count the days from latest release, but this is not possible via pure cmake. So, the second idea is just using number of day in the current year and switch middle number to 0 when new series is started in 1 January (we may do not publish any snapshots after December release).

Comment (gzotti) [reply CMakeLists.txt:45]: Ah, numbers 24.0.* are dev snapshots from January to March! OK.

Comment (alex-w) [reply CMakeLists.txt:45]: Yes, 24.1.* for dev snapshots from March to June...

File: CMakeLists.txt (hunk 3 above)
Comment (gzotti) [reply CMakeLists.txt:54]: It was a signal to discern Qt5- and Qt6-based builds. Is this now done in the package filename?

Comment (alex-w) [reply CMakeLists.txt:54]: > It was a signal to discern Qt5- and Qt6-based builds. Is this now done in the package filename?

Yes for Windows and macOS packages

**Comment by alex-w** [comment]
Is it OK?

**Comment by gzotti** [comment]
SUG 1.2 needs an update. Do I understand correctly that the "internal" number is now history? Can you please fill in your new approach, and I can fix the formulation before merge.

**Comment by alex-w** [comment]
@gzotti please check changes in the SUG

**Comment by gzotti** [comment]
Reading the YY.V.DDD snapshot number, would it not be simper to understand if those were YY.V.MMDD? Month-day numbers (with leading zero where applicable) would also be rising. Else it looks OK for me, although I admit I have not tested actual building!

**Comment by alex-w** [comment]
> Reading the YY.V.DDD snapshot number, would it not be simper to understand if those were YY.V.MMDD? 

The _.DDD_ component is for inner use and for packaging the source code of snapshots - no need extra meaning here.

**Pull request merged**